File: ant-sdk/antd-java/src/main/java/com/autonomi/antd/models/UploadCostEstimate.java
package com.autonomi.antd.models;

/**
 * Pre-upload cost breakdown returned by {@code dataCost} and {@code fileCost}.
 *
 * <p>The server samples up to 5 chunk addresses and extrapolates the storage
 * cost. Gas is an advisory heuristic, not a live gas-oracle query.
 *
 * @param cost                 storage cost in atto tokens (string to preserve precision)
 * @param fileSize             original size in bytes
 * @param chunkCount           number of data chunks the file would split into
 * @param estimatedGasCostWei  advisory gas heuristic in wei (string for precision)
 * @param paymentMode          "auto" | "merkle" | "single"
 */
public record UploadCostEstimate(
    String cost,
    long fileSize,
    int chunkCount,
    String estimatedGasCostWei,
    String paymentMode) {}

File: ant-sdk/antd-php/src/Models/UploadCostEstimate.php
<?php

declare(strict_types=1);

namespace Autonomi\Antd\Models;

/**
 * Pre-upload cost breakdown returned by {@see AntdClient::dataCost()} and
 * {@see AntdClient::fileCost()}.
 *
 * The server samples up to 5 chunk addresses and extrapolates the storage
 * cost. Gas is an advisory heuristic, not a live gas-oracle query.
 */
readonly class UploadCostEstimate
{
    public function __construct(
        public string $cost,                 // storage cost in atto tokens
        public int $fileSize,                // original file size in bytes
        public int $chunkCount,              // number of data chunks
        public string $estimatedGasCostWei,  // advisory gas heuristic in wei
        public string $paymentMode,          // "auto" | "merkle" | "single"
    ) {}
}
